dashboard/server/api/index.js

```diff
@@ -1,7 +1,33 @@
-module.exports = {
-    testSocketConnection: (client) => {
-        client.on('test-conn', (cb) => {
-            cb('hello world');
-        });
+module.exports = (socket) => ({
+    handleNewRun: (data, callback) => {
+        socket.to('frontend').emit('process:start', data);
+        if (callback) {
+            callback('new-run', data);
+        }
```

Comment:
We can improve readability here:

```suggestion
        callback && callback('new-run', data);
```